### PR TITLE
Update contact models

### PR DESCRIPTION
**Pull request by diegoferigo**
Follows up #256 and #260.

----
📚 Documentation preview 📚: https://jaxsim--269.org.readthedocs.build//269/

### CODE REVIEW

**Review by flferretti — APPROVED** [review]
Thanks a lot Diego, that's great! LGTM, I just left some minor comments

File: src/jaxsim/api/model.py [hunk 1]
```diff
@@ -1976,15 +1953,67 @@ def step(
...
+        link_forces=link_forces,
+        joint_force_references=joint_force_references,
+    )
+
+    # =============
+    # Phase 2: step
+    # =============
+
+    # Prepare the references to pass.
+    with references.switch_velocity_representation(data.velocity_representation):
```
[the remark this answers is not in the document]
Comment (diegoferigo) [reply src/jaxsim/api/model.py:2013]: Can you elaborate a bit more this comment? The built `references` object already takes the velocity representation of `data`. The context manager is there to be explicit. Better a line in code mode (that has no overhead) than a possible bug in the future due to changes of the logic above.

File: src/jaxsim/api/contact.py [hunk 2]
```diff
@@ -158,86 +163,58 @@ def collidable_point_dynamics(
...
+                    joint_force_references=joint_force_references,
+                )
+                | kwargs
             )
 
-            aux_data = dict(W_f_avg2_C=W_f̿_Ci, m_tf=m_tf)
-
         case _:
-            raise ValueError(f"Invalid contact model {model.contact_model}")
+            raise ValueError(f"Invalid contact model: {model.contact_model}")
```
[the remark this answers is not in the document]
Comment (diegoferigo) [reply src/jaxsim/api/contact.py]: Done in https://github.com/ami-iit/jaxsim/pull/269/commits/7c78e6458fd8908bcc6ddec28a93900ecc035c8e.

File: src/jaxsim/api/model.py [hunk 3]
```diff
@@ -1976,15 +1953,67 @@ def step(
...
+            jnp.array(
+                [
+                    jnp.logical_not(jnp.allclose(dt, model.time_step)),
+                    jnp.logical_not(
+                        isinstance(
+                            integrator, jaxsim.integrators.fixed_step.ForwardEuler
+                        )
+                    ),
+                ]
+            ).any(),
```
[the remark this answers is not in the document]
Comment (diegoferigo) [reply src/jaxsim/api/model.py]: Done in https://github.com/ami-iit/jaxsim/pull/269/commits/34006fac596e4f0da2ace015c0ed256452b5c4de.

File: src/jaxsim/api/model.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (flferretti) [reply src/jaxsim/api/model.py:2007]: Yes! In line 2003, passing `data` to `JaxSimModelReference.build` will create a `references` object with the same `velocity_representation` of `data`

https://github.com/ami-iit/jaxsim/blob/edc4a35c5e3aa8efb400d0abc08df37db93f7486/src/jaxsim/api/references.py#L98

and we are also passing `velocity_representation` explicitly, so for this reason I believe that the context manager in line 2013 would be unnecessary. Nevertheless, if the intention is to stay on the safe side, then it's more than fine

**Pull request merged**